tests/v2/validation/scaling/nodepools.go

```diff
@@ -0,0 +1,72 @@
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	v1 "github.com/rancher/rancher/tests/framework/clients/rancher/v1"
+	"github.com/rancher/rancher/tests/framework/extensions/machinepools"
+	"github.com/rancher/rancher/tests/framework/extensions/provisioninginput"
+	rke1 "github.com/rancher/rancher/tests/framework/extensions/rke1/nodepools"
+	"github.com/rancher/rancher/tests/framework/extensions/scaling"
+	"github.com/rancher/rancher/tests/framework/extensions/workloads/pods"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	ProvisioningSteveResourceType = "provisioning.cattle.io.cluster"
+	defaultNamespace              = "fleet-default/"
+)
+
+// func TestScalingRKE2K3SNodePools(t *testing.T, client *rancher.Client, cluster *v1.SteveAPIObject, updatedCluster *apisV1.Cluster, machineConfig *v1.SteveAPIObject) {
+func TestScalingRKE2K3SWorkerNodePools(t *testing.T, client *rancher.Client, clusterID string, clustersConfig *provisioninginput.Config) {
+	cluster, err := client.Steve.SteveType(ProvisioningSteveResourceType).ByID(clusterID)
+	require.NoError(t, err)
+
+	updatedCluster := new(apisV1.Cluster)
+	err = v1.ConvertToK8sType(cluster, &updatedCluster)
+	require.NoError(t, err)
+
+	providerName := clustersConfig.Providers[0]
+	nodeProvider := scaling.GetClusterProvider(providerName)
+
+	machineConfig := updatedCluster.Spec.RKEConfig.MachinePools[0].NodeConfig.Name
+	machineConfigResp, err := client.Steve.SteveType(nodeProvider.MachineConfigPoolResourceSteveType).ByID(defaultNamespace + machineConfig)
+	require.NoError(t, err)
+
+	err, createdClusterResp, createdNodePoolCluster := machinepools.CreateNewWorkerMachinePool(client, cluster, updatedCluster, machineConfigResp, 3)
+	require.NoError(t, err)
+
+	err = pods.VerifyReadyDaemonsetPods(t, client, cluster)
+	require.NoError(t, err)
+
+	err, scaledClusterResp := machinepools.ScaleNewWorkerMachinePool(client, createdClusterResp, createdNodePoolCluster, 4)
+	require.NoError(t, err)
+
+	err = pods.VerifyReadyDaemonsetPods(t, client, cluster)
+	require.NoError(t, err)
+
+	err, scaledClusterResp = machinepools.ScaleNewWorkerMachinePool(client, createdClusterResp, createdNodePoolCluster, 3)
+	require.NoError(t, err)
+
+	err = pods.VerifyReadyDaemonsetPods(t, client, cluster)
+	require.NoError(t, err)
+
+	err = machinepools.DeleteWorkerMachinePool(client, cluster, scaledClusterResp, createdNodePoolCluster)
+	require.NoError(t, err)
+}
+
+func TestScalingRKE1WorkerNodePools(t *testing.T, client *rancher.Client, clusterID string, clustersConfig *provisioninginput.Config) {
+	providerName := clustersConfig.Providers[0]
+	nodeProvider := scaling.GetRKE1ClusterProvider(providerName)
+
+	nodeTemplate, err := nodeProvider.NodeTemplateFunc(client)
+	require.NoError(t, err)
+
+	cluster, err := client.Management.Cluster.ByID(clusterID)
+	require.NoError(t, err)
+
+	err = rke1.ScaleNewWorkerNodePool(client, cluster, nodeTemplate)
```

Comment:
I had to look into this function to know what exactly was being scaled. I feel this should be in the test function, not in the 'scaleNewWorkerNodePool' function. This will also allow you to scale up and down (see comments in the rke1/nodepools file)

Reply:
This initially was in the test function, but with the refactoring changes, I was under the impression that we don't want to follow that practice anymore.

Reply:
Reworked this so that it is on parity with RKE2/K3S. As such, resolving comment.